pkg/cloud/nodelocal/nodelocal_storage.go

```diff
@@ -213,41 +211,8 @@ func (*localFileStorage) Close() error {
 	return nil
 }
 
-var _ externalconn.ConnectionDetails = &localFileConnectionDetails{}
```

Comment:
note for reviewers: I just moved some of these methods to `nodelocal_connection.go` to keep all ExternalConnection stuff in one place. `Dial()` is the only new method.